(RelGen) C:\Users\admin\Desktop\第六章数据发布实验\tools\RelGen-main>python -u "c:\Users\admin\Desktop\第六章数据发布实验\tools\RelGen-main\quick_test.py"
Model Initialization
Diffusion Initialization
Step 500/500 Loss: 0.9518
training time: 133.8441767692566
Sample timestep    0
Traceback (most recent call last):
  File "D:\Anaconda3\envs\RelGen\lib\site-packages\numpy\core\fromnumeric.py", line 57, in _wrapfunc
    return bound(*args, **kwds)
TypeError: clip() got an unexpected keyword argument 'out'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "c:\Users\admin\Desktop\第六章数据发布实验\tools\RelGen-main\quick_test.py", line 43, in <module>
    sampled_data = synthesizer.sample(device=torch.device("cuda"))
  File "c:\Users\admin\Desktop\第六章数据发布实验\tools\RelGen-main\relgen\synthesizer\diffusionsynthesizer\relddpm_synthesizer.py", line 197, in sample
    full_sampled_data = self.dataset.tables[join_table_name].inverse(full_sampled_data)
  File "c:\Users\admin\Desktop\第六章数据发布实验\tools\RelGen-main\relgen\data\table.py", line 134, in inverse
    inverse_data.append(column.inverse(col_data))
  File "c:\Users\admin\Desktop\第六章数据发布实验\tools\RelGen-main\relgen\data\column.py", line 355, in inverse
    cat_data = self.index_to_cat(data_instance)
  File "c:\Users\admin\Desktop\第六章数据发布实验\tools\RelGen-main\relgen\data\column.py", line 256, in index_to_cat
    values = np.clip(data_instance, 0, len(self.distinct_values)-1)
  File "<__array_function__ internals>", line 6, in clip
  File "D:\Anaconda3\envs\RelGen\lib\site-packages\numpy\core\fromnumeric.py", line 2115, in clip
    return _wrapfunc(a, 'clip', a_min, a_max, out=out, **kwargs)
  File "D:\Anaconda3\envs\RelGen\lib\site-packages\numpy\core\fromnumeric.py", line 66, in _wrapfunc
    return _wrapit(obj, method, *args, **kwds)
  File "D:\Anaconda3\envs\RelGen\lib\site-packages\numpy\core\fromnumeric.py", line 43, in _wrapit
    result = getattr(asarray(obj), method)(*args, **kwds)
  File "D:\Anaconda3\envs\RelGen\lib\site-packages\torch\tensor.py", line 630, in __array__
    return self.numpy()
TypeError: can't convert cuda:0 device type tensor to numpy. Use Tensor.cpu() to copy the tensor to host memory first.